src/main/java/com/google/sps/servlets/AddFriendsServlet.java

```diff
@@ -0,0 +1,86 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import com.google.sps.data.Profile;
+import com.google.gson.Gson;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.FetchOptions;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import org.json.simple.JSONObject;
+
+import java.util.Arrays;
+import java.util.ArrayList;
+import java.util.List;
+
+/** Servlet that posts and gets user profiles in Datastore */
+@WebServlet("/addFriend")
+public class AddFriendsServlet extends HttpServlet {
+  
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String friendEmail = request.getParameter("friendEmail");
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    Query query = new Query("Profile")
+      .setFilter(new Query.FilterPredicate("email", Query.FilterOperator.EQUAL, friendEmail));
+    PreparedQuery results = datastore.prepare(query);
+    Entity entity = results.asSingleEntity();
+    if(entity != null) {
+      UserService userService = UserServiceFactory.getUserService();
+      if (!userService.isUserLoggedIn()) {
+        response.setContentType("application/html");
+        response.getWriter().println("Please log in to add friends");
+      } else {
+        query = new Query("Profile")
+          .setFilter(new Query.FilterPredicate("id", Query.FilterOperator.EQUAL, userService.getCurrentUser().getUserId()));
+        results = datastore.prepare(query);
+        entity = results.asSingleEntity();
+        if(entity == null) {
+          response.setContentType("application/html");
+          response.getWriter().println("Please create your user profile before adding friends");
+        } else {
+          ArrayList<String> friendList = (ArrayList<String>) entity.getProperty("friendList");
+          if(friendList.contains(friendEmail)){
+            response.setContentType("application/html");
+            response.getWriter().println("You already have this user as a friend");
+          } else {
+            friendList.add(friendEmail);
+            entity.setProperty("friendList", friendList);
+            datastore.put(entity);
+            response.setContentType("application/html");
+            response.getWriter().println("Successfully added");
+          }
+        }
+      }
+    }
+    else {
```

Comment:
} else { in one line

Also, you might want to move the else block to the top, i.e. if (entity == null) {} else {}